flink/v1.19/flink/src/main/java/org/apache/iceberg/flink/maintenance/operator/TableChange.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.flink.maintenance.operator;
+
+import java.util.Objects;
+import org.apache.iceberg.DataFile;
+import org.apache.iceberg.DeleteFile;
+import org.apache.iceberg.Snapshot;
+import org.apache.iceberg.io.FileIO;
+import org.apache.iceberg.relocated.com.google.common.base.MoreObjects;
+
+/** Event describing changes in an Iceberg table */
+class TableChange {
```

Comment:
how will this be used by downstream operators for planning purpose?

also how does this work with maintenance tasks like snapshot expiration?

Reply:
This will be used for Maintenance Task scheduling as described in the [design doc](https://docs.google.com/document/d/16g3vR18mVBy8jbFaLjf2JwAANuYOmIwr15yDDxovdnA). The `Scheduling` paragraph describes the require parameters of the `TableChange` object. Also the `Monitor` paragraph adds some more details.

Reply:
you said the purpose of `MonitorSource` and `TableChange` are to provide input to `Trigger` operator so that the `Trigger` code.

Here are the trigger modes discussed before

1. standalone streaming/long-running compaction job. This will use the `MonitorSource` to emit `TableChange` to `Trigger` operator, right?
2. embedded trigger mode in post commit stage of a write job. I assume post commit can emit the `TableChange` event.
3. external trigger mode (like Airflow). how is the Trigger input calculated in this case?

Reply:
> 1. standalone streaming/long-running compaction job. This will use the `MonitorSource` to emit `TableChange` to `Trigger` operator, right?

Yes, you're right 

> 2. embedded trigger mode in post commit stage of a write job. I assume post commit can emit the `TableChange` event.

We need a simple mapper, I have called it `Commit Converter` in the doc

> 3. external trigger mode (like Airflow). how is the Trigger input calculated in this case?

In this case the full sceduling is on the external trigger.
I expect that this will be based mostly on timers, but using the Java or Python API is a possibility too